tests/unit/fem/test_fe_compatibility.cpp

```diff
@@ -0,0 +1,1302 @@
+// Copyright (c) 2010-2025, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "mfem.hpp"
+#include "unit_tests.hpp"
+
+using namespace mfem;
+
+Element * GetElement(Geometry::Type type)
+{
+   Element *el = NULL;
+   switch (type)
+   {
+      case Geometry::POINT:
+         el = new Point;
+         break;
+      case Geometry::SEGMENT:
+         el = new Segment;
+         break;
+      case Geometry::TRIANGLE:
+         el = new Triangle;
+         break;
+      case Geometry::SQUARE:
+         el = new Quadrilateral;
+         break;
+      case Geometry::TETRAHEDRON:
+         el = new Tetrahedron;
+         break;
+      case Geometry::CUBE:
+         el = new Hexahedron;
+         break;
+      case Geometry::PRISM:
+         el = new Wedge;
+         break;
+      case Geometry::PYRAMID:
+         el = new Pyramid;
+         break;
+      default:
+         break;
+   }
+   return el;
+}
+
+// Build a mesh containing a single element
+Mesh MakeElementMesh(Geometry::Type type, real_t * vertices)
+{
+   Element *elem = GetElement(type);
+
+   int nvert = elem->GetNVertices();
+
+   Array<int> el_inds(nvert), el_attr(1);
+   for (int i=0; i<nvert; i++) { el_inds[i] = i; }
+   el_attr[0] = 1;
+
+   int dim = 0, sdim = -1;
+   Geometry::Type bdr_type = Geometry::INVALID;
+   if (type == Geometry::SEGMENT)
+   {
+      dim = 1;
+      sdim = 1;
+      bdr_type = Geometry::POINT;
+   }
+   else if (type >= Geometry::TRIANGLE && type <= Geometry::SQUARE)
+   {
+      dim = 2;
+      sdim = 2;
+      bdr_type = Geometry::SEGMENT;
+   }
+   else if (type >= Geometry::TETRAHEDRON)
+   {
+      dim = 3;
+      sdim = 3;
+      bdr_type = Geometry::TRIANGLE;
```

Comment:
I don't get why the `bdr_type` is always a triangle. How about cubes, wedges, and pyramids?

Reply:
That argument is actually not used because I'm not creating any boundary elements in these meshes. I should probably leave it as ` Geometry::INVALID` to avoid confusion. I'll double check that this would work before making the change.

Reply:
It works fine a `Geometry::INVALID`. Thanks for catching this!